### PR TITLE
OCLOMRS-445: Fix pagination on Add CIEL concepts page when datatype/class filters are selected.

### PR DESCRIPTION
**Pull request by AGMETEOR**
# JIRA TICKET NAME:
[Fix pagination on Add CIEL concepts page when datatype/class filters are selected.](https://issues.openmrs.org/browse/OCLOMRS-445)

# Summary:
On the Add CIEL concepts page;
- Once a user selects concept by datatype or class.
 * The pagination buttons at the right bottom corner of the table stop functioning.
 * That is they change the pagination number but the concepts being displayed do not change.

 * Fix the pagination by ensuring that a user can select a  specific datatype/class and view the next and previous page when they click the pagination buttons.

### CODE REVIEW

File: src/tests/bulkConcepts/containers/BulkConceptsPage.test.js [hunk 1]
```diff
@@ -333,6 +334,38 @@ describe('Test suite for BulkConceptsPage component', () => {
...
+      addToFilterList: jest.fn(),
+      fetchFilteredConcepts: jest.fn(),
+      addConcept: jest.fn(),
+      previewConcept: jest.fn(),
+      handleNextPage: jest.fn(),
+      componentDidUpdate: jest.fn(),
+    };
+    const wrapper = shallow(<BulkConceptsPage {...props} />).setState({ filterOn: true });
+    const instance = wrapper.instance();
+    expect(instance.componentDidUpdate(2)).toEqual(undefined);
```
Comment (dkayiwa) [inline src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: How is this test related to the pagination fix?

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: @dkayiwa , yes, it's related.

Comment (dkayiwa) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: How?

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: The test is similar to the one just above it for a normal search and then pagination only that for this particular one, filtering in true in the state.

Comment (dkayiwa) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: So does this test fail when i set filtering to false?

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: No, it doesn't. Let me modify the test or take it out.

Comment (dkayiwa) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: Then it is not testing what it should.

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: @dkayiwa I think I made a mistake on using the test just before this one as a template. I have now improved this particular test.

File: src/tests/bulkConcepts/containers/BulkConceptsPage.test.js [hunk 2]
```diff
@@ -333,6 +334,39 @@ describe('Test suite for BulkConceptsPage component', () => {
...
+      addToFilterList: jest.fn(),
+      fetchFilteredConcepts: jest.fn(),
+      addConcept: jest.fn(),
+      previewConcept: jest.fn(),
+      handleNextPage: jest.fn(),
+    };
+    const wrapper = shallow(<BulkConceptsPage {...props} />);
+    const instance = wrapper.instance();
+    wrapper.setState({ filterOn: true });
+    instance.componentDidUpdate(props.currentPage + 1);
```
Comment (dkayiwa) [inline src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: Will this test fail if the above page is not incremented?

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: No, it will not. What the test actually looks for is whether an update is made in the component and then uses the current page to fetch the concepts. Let me change the description so that it's not misleading in terms of what the test does.

Comment (dkayiwa) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: https://en.wikipedia.org/wiki/Test-driven_development

Comment (AGMETEOR) [reply src/tests/bulkConcepts/containers/BulkConceptsPage.test.js]: I updated the test to remove the increment in the current page props.
